test/HHConfiguration.py

```diff
@@ -52,22 +49,24 @@
             minDR_l_j_Cut = cms.untracked.double(0.3),
             hltDRCut = cms.untracked.double(0.3),
             hltDPtCut = cms.untracked.double(0.5)  # cut will be DPt/Pt < hltDPtCut
-            ),
+            )
         )
     )
+
+# Add PUPPI MET
+puppiCfg = cms.PSet(METProducer.default_configuration.clone())
+puppiCfg.prefix = cms.string('puppimet_')
+puppiCfg.parameters.met = cms.untracked.InputTag('slimmedMETsPuppi')
+framework.addProducer('puppimet', puppiCfg)
+
 #framework.redoJEC()
+framework.smearJets()
 framework.doSystematics(['jec', 'jer'])
 process = framework.create()
 
-# Add PUPPI MET
-process.framework.producers.puppimet = cms.PSet(METProducer.default_configuration.clone())
-process.framework.producers.puppimet.prefix = cms.string('puppimet_')
-process.framework.producers.puppimet.parameters.met = cms.untracked.InputTag('slimmedMETsPuppi')
-
-# Custom the cuts
+# If we want to modify fwk parameters, it has to be done after create()
```

Comment:
which means `framework.smearJets()` should be after `process = framework.create()` ?